lib/chef/knife/azure_server_create.rb

```diff
@@ -253,6 +253,14 @@ def load_winrm_deps
         :long => "--winrm-max-memoryPerShell MB",
         :description => "Set winrm max memory per shell in MB"
 
+
+      option :delete_chef_config,
+        :long => "--delete-chef-config",
+        :boolean => true,
+        :default => false,
+        :description => "Set this flag to delete chef configuration files during chef extension uninstall or update process it by default set to false"
```

Comment:
how about "Determines whether Chef configuration files when Azure removes the Chef resource extension from the VM. This option is only valid for the cloud-api bootstrap protocol. The default is false."